packages/react-data-grid/src/Viewport.tsx

```diff
@@ -64,19 +57,17 @@ export default function Viewport<R>({
   overscanRowCount,
   overscanColumnCount,
   enableIsScrolling,
+  viewportWidth,
   ...props
 }: ViewportProps<R>) {
-  const viewport = useRef<HTMLDivElement>(null);
   const resetScrollStateTimeoutId = useRef<number | null>(null);
-  const [scrollState, setScrollState] = useState<ScrollState | null>(null);
+  const [scrollState, setScrollState] = useState<ScrollState>({
+    scrollLeft: 0,
+    scrollTop: 0,
+    scrollDirection: SCROLL_DIRECTION.NONE
```

Comment:
I'd rather split those into 3 states, so we can potentially avoid re-renders if we call `setX` with the same value.

Reply:
Done